feat: Update eslint+webpack

yonabystedt opened the pull request:
- update to eslint 9 and new config structure
- projects can supply their own webpack config to be merged with our defaults

scottander commented on vertigis-web-sdk-0.0.1-semantically-released.tgz:
Is this supposed to be checked in?

yonabystedt replied:
Nope :)

yonabystedt commented:
>  I think it is probably worth trying to get this PR updated, to test updates etc.

@scottander I tested the upgrade script and it looks like it works fine, with the caveat about eslint breaking as detailed above. Do you think it's work trying to enhance the script to fix this automatically? It's pretty simple right now, you can get more or less the same results by just updating the `web` and `web-sdk` packages in package.json and running `npm i`.

One "extra" thing it does though is kind of weird? It dumps all the dependencies of Web out as explicit devDependencies. This is explained in a comment:

```
// Add or update all of Web's dependencies as dev dependencies of this
// project. Although this is not strictly necessary for the project to
// build, aspects of VS Code intellisense like auto imports will not
// work without this
```

Very interesting! This seems to be the solution to the auto imports issue that keeps cropping up. However, why would we only do this on upgrade and not initially? It seems like we should consistently do it one way or the other :/

yonabystedt commented:
> Very interesting! This seems to be the solution to the auto imports issue that keeps cropping up. However, why would we only do this on upgrade and not initially? It seems like we should consistently do it one way or the other :/

After doing some testing I found that not only does it not help with auto imports, it actually breaks them further by offering auto imports we don't want from `react-ui` and `mui`. Weirdly the same ones don't work, and they consistently don't work in all packages.

As such, I've removed this code from the upgrade script. We don't want to add these packages to SDK projects, especially without warning.

yonabystedt commented:
I added some capabilities to the upgrade script that let it upgrade old projects to use the new eslint configuration. Because this is now taken care of by the upgrader I've also removed the old `.eslintrc.js` config that was being kept around for compatibility.

yonabystedt commented:
@lfan2020 Do you have experience testing this repo? It basically just needs a once over to make sure that everything is working properly after so many changes :)

There's a couple of new things the upgrade script should be doing too. It should change in the `package.json` of an SDK project the property `type` from `commonjs` to `module`, and it should also add a file called `eslint.config.js` to the root of the project.

Thanks!

yonabystedt commented:
Moved to a new branch
